src/adapter.js

```diff
@@ -3,92 +3,112 @@
 const { filter, sortAll, take, map } = require('./utils')
 const drain = require('it-drain')
 
-class InterfaceDatastoreAdapter {
-  async open () { // eslint-disable-line require-await
-
+/**
+ * @typedef {import('./key')} Key
+ * @typedef {import('./types').Pair} Pair
+ * @typedef {import('./types').Datastore} Datastore
+ * @typedef {import('./types').Options} Options
+ * @typedef {import('./types').Query} Query
+ * @typedef {import('./types').Batch} Batch
+ */
+
+/**
+ * @template O
+ * @typedef {import('./types').AwaitIterable<O>} AwaitIterable
+ */
+
+/**
+ * @implements {Datastore}
+ */
+class DatastoreBase {
```

Comment:
I would either rename the file this is in, or don't rename the class.  The inconsistency of having one thing called an `adapter` and another called a `base` is confusing.